### DISCUSSION TITLE
Project setup

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#4] [General]
First up, how do we setup the solution:

- Single project
- Two projects (web and core)
- Other

**Comment by CarlSargunar** [comment]
I alwys prefer a 2 project setup as a predominantly back end dev, and try and structure all my projects that way. Consciouis that it's a barrier to entry to front end devs, but I would be curious if there was a poll for how many project were set up outside of VS, and with a single project

I mean is anyone downloading the zip for Umbraco and setting it up that way anymore?

**Comment by NikRimington** [comment]
Personally I think 2 projects, Web and Core, is better as it encourages better separation and potential for unit testing (not that the latter is likely to part of this concept)

Another option I've seen is 3, Web, Core, FE, where the FE project contains template pages css, js build tasks etc and the outputs are copied into the Web.

**Reply by CarlSargunar** [reply]
I've seen 3 too, but I've also seen a lot of different tooling for copying the FE assets to the web project - if we can achieve that in a univesally accepted way (and I'm not the best person for that as I'm strictly BE) then I think 3 works, but I'd be happier with 2

**Comment by Mantus667** [comment]
I mostly setup everything in 3 projects. One Core, one web and one separate project where the ModelsBuilder files are stored. Find this is a neat way of handling the models.

As Nik suggests you could also have a separate project for the frontend stuff in regards to css/js. Which is something I also encountered. This then could be bundled with Gulp. The new package template and I think the Umbraco source are using gulp for the angularjs stuff. As I'm also a backend dev I hope someone with more knowdledge about frontend willl step up here.

**Reply by TimGeyssens** [reply]
hmm about the frontend stuff, is it an idea that we keep this seperated... not in vs? Since then it can work cross platform...

**Reply by TimGeyssens** [reply]
what is the advantage of a seperate models builders file project?

**Reply by Mantus667** [reply]
I just like to have the models on disk so I can check it into source control. Especially if I extend them.
You could argue that it would also make sense to put them into the Core project.
I just find it nicer to have something which is autogenerated in a separate project.
So there are no special reasons to do it this way. For me, it's just a cleaner setup. :)

**Reply by CarlSargunar** [reply]
I tend to have them in the core project exactly for this reason - we frequently extend them and havng them in the same project as services etc makes it easier to work with IMO

**Reply by TimGeyssens** [reply]
yeah on disk! defo

**Comment by dawoe** [comment]
I think for a starter kit having a core and web should do. 
But personally I prefer to have them seperated by feature. A bit like Callum talks about here : https://24days.in/umbraco-cms/2018/reducing-tight-coupling/

**Reply by Mantus667** [reply]
Haven't seen that article, so thanks for that. :)
Moving everything into feature projects is a bit of a big task for this or?
But at least I prefer to use feature folders in my projects so everything is at least in one place. I guess that is one step before actually moving to feature projects.

Interestingly he also moves the generated models into it's own project the way I do it.
